### ISSUE TITLE
How to update only the form that changed when multiple forms exists in same page

### ISSUE THREAD

**Issue by chbdetta** [erikras/redux-form#544]
I have like 5 redux-form component in a single page, when a field in one of those forms changes, all of the 5 forms update, causing very bad performance.
I try to use `shouldComponentUpdate` to limit the updating to the form that actually changed but I don't find any information from the `props` passed by redux form.

`pristine` denotes if the form is different from its initial state but I want to have some flag that indicates if some of the form's field is different from last change.

**Comment by erikras** [comment]
If they are all attached to the same form (i.e. the `form` config parameter is the same for all of them), they must necessarily all update because they are listening to that slice of the Redux state.

If they are connected to different forms, [they will not all re-render](https://github.com/erikras/redux-form/blob/master/src/__tests__/createReduxForm.spec.js#L1732-L1782).

And [only the field that changes will mutate](https://github.com/erikras/redux-form/blob/master/src/__tests__/createReduxForm.spec.js#L1707-L1730), so you can use `===` on each field when deciding whether or not to rerender the input.

**Comment by dbismut** [comment]
Hey @erikras. Thanks for this incredible package.
It's probably me doing something wrong, but I seem to have an equivalent problem.

I have a contactForm always present on my page. Everything is working fine, but whenever the route change, that contactForm is re-rendered.

I did a quick equality test in `componentDidMount` with `for(key in prevProps) { console.log(key, prevProps[key] === this.props[key]) }`, and here is the result:

```javascript
submitPassback: false
dispatch: true
__esModule: true
readonly: true
active: true
asyncValidating: true
dirty: true
error: true
errors: true
fields: true
formKey: true
invalid: true
pristine: true
submitting: true
submitFailed: true
valid: true
values: true
asyncValidate: false
destroyForm: false
handleSubmit: true
initializeForm: false
resetForm: false
touch: false
touchAll: false
untouch: false
untouchAll: false
```

So it looks a lot like the functions passed to the contactForm don't pass the equality test and therefore trigger a re-render. Is this normal behavior?

Thanks!

**Comment by erikras** [comment]
@dbismut Fantastic analysis. :+1: That's very helpful. I think you've found a bug.

**Comment by dbismut** [comment]
Hi @erikras, any chance this can be resolved easily or are you waiting for 6.0.0 to start from scratch?

**Comment by danielrob** [comment]
Closing with bulk close of v5 issues. If this issue still requires action and/or also affects v6/7, please request a reopen with details or raise a new issue. v5 PR's are still welcomed.
